### PR TITLE
PR: Update user-facing Changelog for Spyder 6.0

### PR DESCRIPTION
**Pull request by ccordoba12**
## Description of Changes

- Add to Changelog the most important improvements from the last alpha and beta releases.
- Also, add instructions necessary to release a new major version.

### Issue(s) Resolved

Fixes #

### Affirmation

By submitting this Pull Request or typing my (user)name below,
I affirm the [Developer Certificate of Origin](https://developercertificate.org)
with respect to all commits and content included in this PR,
and understand I am releasing the same under Spyder's MIT (Expat) license.

I certify the above statement is true and correct: @ccordoba12

### CODE REVIEW

**Comment by ccordoba12** [comment]
@dalthviz, please check the release instructions I added and let me know if they are clear enough for you.

File: RELEASE.md [hunk 1]
```diff
@@ -152,6 +152,14 @@ To release a new version of Spyder you need to follow these steps:
 
 * git pull or git fetch/merge the respective branch that will be released (e.g `6.x` - stable branch or `master` - alphas or betas of a new major version).
 
+* For a new major release (e.g. version 6.0.0 after 5.5.6):
+
+    - `git checkout -b 6.x`
+    - `git checkout master`
+    - Update major version in `__init__.py` to `7`.
```
Comment (dalthviz) [inline RELEASE.md:159]: Maybe this should say something like:
```suggestion
    - Update version in `__init__.py` to reflect next major version as dev version (i.e `7.0.0.dev0`).
```

**Review by dalthviz — COMMENTED** [review]
Thanks @ccordoba12 ! Left a suggestion about the instruction related with bumping to the next major version (`7.0.0.dev0` when doing the `6.0.0` release for example). Other than that I think the added instructions make sense :+1: Just to confirm, basically those instructions aim to create a new branch from where the new major release should be done (Spyder 6 - `6.x` branch) while also updating the master branch version to show the current next major release being developed (Spyder 7 - `master` branch),  right?

**Comment by ccordoba12** [comment]
> Just to confirm, basically those instructions aim to create a new branch from where the new major release should be done (Spyder 6 - 6.x branch) while also updating the master branch version to show the current next major release being developed (Spyder 7 - master branch), right?

Yep, that's the idea. Before starting the final part of a major release, we need to create a branch for the new stable Spyder version. And bumping the major version in `master` creates a difference between that branch and `master` moving forward.

**Review by dalthviz — APPROVED** [review]
Thanks @ccordoba12 !

**Pull request merged**